### PR TITLE
feat: curio forest compat

### PR DESCRIPTION
**Pull request by snadrus**
## Related Issues

Limited the API set in Curio to this list. 

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Comment by snadrus** [comment]
Along with the given list, Curio also needed:
- StateMinerActiveSectors
- MpoolPushMessage
- StateSectorPartition
Do we need to tell Forest about this? @LexLuthr

File: node/modules/storageminer.go [hunk 1]
```diff
@@ -147,7 +148,7 @@ func StorageNetworkName(ctx helpers.MetricsCtx, a v1api.FullNode) (dtypes.Networ
 	return a.StateNetworkName(ctx)
 }
 
-func SealProofType(maddr dtypes.MinerAddress, fnapi v1api.FullNode) (abi.RegisteredSealProof, error) {
+func SealProofType(maddr dtypes.MinerAddress, fnapi daemonapi.Daemon) (abi.RegisteredSealProof, error) {
```
Comment (LexLuthr) [inline node/modules/storageminer.go:151]: Let's copy over this function to Curio deps to fix the failing tests. I think that's the cheapest solution here instead is a complicated FX dependency fix.

**Review by LexLuthr — CHANGES_REQUESTED** [review]
We have a usage in `chain-connectivity.mjs` as well to track chain sync. We might need to modify that to work individually with node types. We can probably handle that in the same PR where we add the Forest integration.

**Comment by LexLuthr** [comment]
Closing as this has been ported.

**Pull request closed without merging**